app/src/androidTest/java/org/resolvetosavelives/red/newentry/search/PatientRepositoryTest.kt

```diff
@@ -51,6 +51,52 @@ class PatientRepositoryTest {
     assertThat(search2).isNotEmpty()
   }
 
+  @Test
+  fun createAnOngoingPatientEntry_withNullDateOfBirth_andAgeFilledIn_thenSaveItCompletely() {
+    val ongoingAddress = OngoingPatientEntry.Address("Noida", "Gautam Buddha Nagar", "Uttar Pradesh")
+    val ongoingPersonalDetails = OngoingPatientEntry.PersonalDetails("Ashok Kumar", null, "42", Gender.MALE)
+
+    val personalDetailsOnlyEntry = OngoingPatientEntry(personalDetails = ongoingPersonalDetails)
+
+    patientRepository.saveOngoingEntry(personalDetailsOnlyEntry)
+        .andThen(patientRepository.ongoingEntry())
+        .map { ongoingEntry -> ongoingEntry.copy(address = ongoingAddress) }
+        .flatMapCompletable { withAddress -> patientRepository.saveOngoingEntry(withAddress) }
+        .andThen(patientRepository.saveOngoingEntryAsPatient())
+        .subscribe()
+
+    val search1 = patientRepository.searchPatients("lakshman")
+        .take(1)
+        .blockingFirst()
+
+    assertThat(search1).isEmpty()
+
+    val search2 = patientRepository.searchPatients("ashok")
+        .take(1)
```

Comment:
I just realized that `take(1)` is redundant when we are already using `blockingFirst()`.